File: java/src/main/java/deltix/luminary/DecoratorPropertyValue.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;

public class DecoratorPropertyValue {
    private final DecoratorPropertyDef definition;
    private final boolean isDefault;
    private Literal value;

    public DecoratorPropertyValue(@NotNull DecoratorPropertyDef definition, @NotNull Literal value, boolean isDefault) {
        this.definition = definition;
        this.value = value;
        this.isDefault = isDefault;
    }

    @NotNull
    public DecoratorPropertyDef getDefinition() {
        return definition;
    }

    @NotNull
    public Literal getValue() {
        return value;
    }

    public boolean isDefault() {
        return isDefault;
    }

    void setValue(Literal value) {
        this.value = value;
    }
}

File: csharp/main/DecoratorPropertyValue.cs
using System;

namespace Deltix.Luminary
{
	public sealed class DecoratorPropertyValue
	{
		public DecoratorPropertyValue(DecoratorPropertyDef definition, Literal value, Boolean isDefault)
		{
			Definition = definition;
			Value = value;
			IsDefault = isDefault;
		}

		public DecoratorPropertyDef Definition { get; }

		public Literal Value { get; internal set; }

		public Boolean IsDefault { get; }
	}
}
